interview-questions.md

```diff
@@ -7,44 +7,93 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Your answer:
+  Your answer: I would begin by definining what a foreign key is which is known a column that connects two data keys in a table of data within a relaitonal database. In other words, a foriegn key acts almost like a holding of data that references a primary key in  the orginal table. I would fix this problem by  simply adding the foreign key by updating the attributes within a column. 
 
   Researched answer:
+I can do what I mentioned previously about adding a new column but I would identify this column by adding and integer to the cohort_id:integer
+I can rails db:migrate in rails app cohort.rb
+go to has_many students  and then add student model cohoty_id: integer
 
+I can also restart everything and add my foriegn key fromt he start but that would be a waste time. Migrating would be best.
 
 
 2. Which RESTful routes must always be passed params? Why?
 
   Your answer:
+edit - in order for something to be editing we have to pass it rhough pararms to know exxaclty what is being edited in the [id] we jsut worked on this int he rails-commenting.rb assessment
+find - when we find something we are looking at the url attached to the what we are finding that has it own id that need params.
+destroy - in order to destory a particular id we need to know what we are destroying with params
 
   Researched answer:
 
+All three routes listed above need to have params so that we can find the key we are refering to
+so that we can perform these tasks. 
+
+Here are three examples of these routes from our assessment to give you a better understadning on  how without params we can perform the data routes: 
+
+def edit
+    # ---6)def edit method is calling on @posts to find one primary id and displays one id
+    @post = BlogPost.find(params[:id])
+  end
 
+def destroy
+    @post = BlogPost.find(params[:id])
+    if @post.destroy
+      redirect_to blog_posts_path
+    else
+      # ---8)we are deleting, destroying and removing data from our @posts. We are sayign that if our post can not be delted then we must redirect to another path to our @post.
+
+
+Update and create routes also need params.
 
 3. Name three rails generator commands. What is created by each?
 
   Your answer:
 
+rails db:create - Is when you are creating database in rails you first input this command in the terminal and it creates the database you will work on
+rails s - you input rails s in the terminal and  you are running your rails server allowing your capabaility to see your rails app go live  on locahhost3000.
+rails db:migrate - it runs the single migrations you have not run yet. It updates the data base.
+
+
   Researched answer:
 
+rails db:create - creates the database
+rails s - begins the rails server by running the rails app
+rails db:migrate - this command updates different versions of your database to accomplish your task at different times. It updates the schema file that holds our created data that we can update, create, or destory data in.
```

Comment:
Generate commands are a little different than these rails commands.  A rails generate command will create files in our rails app.  For instance $ rails generate Model